x-pack/plugins/file_upload/public/components/geojson_upload_form/index_name_form.tsx

```diff
@@ -15,8 +15,8 @@ export interface Props {
   indexName: string;
   indexNameError?: string;
   onIndexNameChange: (name: string, error?: string) => void;
-  onIndexNameValidationStart: () => void;
-  onIndexNameValidationEnd: () => void;
+  onIndexNameValidationStart?: () => void;
```

Comment:
onIndexNameValidationStart and onIndexNameValidationEnd should be required so that there is user feedback that the name is getting validated and users should not be able to click the next button.

Reply:
I don't believe these should be required. Whether a calling plugin decides to execute an action on either of these events should be up to that plugin. I agree we can make a case for using them in our scenario but these shouldn't be mandatory

Reply:
my2c I'd remove the optionality. It's not needed, since all consumers _do_ use it, and it will remove the need for additional handling missing params.  From experience, optional params tend to sneak in in our APIs over time, which just increases the surface area of gaps that needed to be handled on the caller's end. So I'd avoid it wherever we can.

Reply:
am making these required in https://github.com/elastic/kibana/pull/100278. imho no need to be optional